src/main/scala/Server.scala

```diff
@@ -1,16 +1,18 @@
-import repo.Repository.InMemoryCustomerRepository
 import cats.effect._
-import endpoints.CustomerRoutes.customerRoutes
+import domain.CustomerService
+import gateway.CustomerRoutes.customerRoutes
+import infrastructure.repositories.InMemoryCustomerRepository
 import org.http4s.HttpApp
 import org.http4s.blaze.server.BlazeServerBuilder
 import org.http4s.server.Router
 import org.http4s.implicits._
 
 object Server extends IOApp {
   private val customerRepo = new InMemoryCustomerRepository
+  val customerService      = new CustomerService[IO](customerRepo)
 
   private val httpApp: HttpApp[IO] = Router(
```

Comment:
the only part where u can specify `IO` is top construct - so here it is OK, but everywhere else - u need abstracted `F` (tagless final approach)